mne_bids/read.py

```diff
@@ -328,7 +328,117 @@ def _handle_channels_reading(channels_fname, bids_fname, raw):
     return raw
 
 
-def read_raw_bids(bids_fname, bids_root, extra_params=None,
+def _infer_kind(*, bids_basename, bids_root, sub, ses):
+    # Check which kind is available for this particular
+    # subject & session. If we get no or multiple hits, throw an error.
+
+    kinds = _get_kinds_for_sub(bids_basename=bids_basename,
+                               bids_root=bids_root, sub=sub, ses=ses)
+    if not kinds:
+        raise ValueError('No data found.')
+    elif len(kinds) >= 2:
```

Comment:
this is not good. What if you have `beh` and `meg`? You should read only `meg`

Reply:
maybe you should check if, amongst `meg`, `eeg`, and `ieeg`, there are two. I think this is a case you will rarely hit but good to have it.

Reply:
Addressed in f51a3f8